sphinxcontrib/needs/layout.py

```diff
@@ -57,13 +57,13 @@ def create_need(need_id, app, layout=None, style=None, docname=None):
     node_inner.attributes['ids'].append(need_id)
 
     if layout is None:
-        if need_data['layout'] is None or len(need_data['layout']) == 0:
+        if need_data['layout']:
```

Comment:
there's a bug here, which is related to the failing CI.

I seem to have uncovered it by cleaning up the boolean logic above.

i'll add some comments to the original code to illustrate what i mean here

```python
if layout is None:
    if need_data['layout'] is None or len(need_data['layout']) == 0  # this is equivalent to `if need_data['layout']`
        layout = getattr(app.config, 'needs_default_layout', 'clean')
    else:  # need_data['layout'] is 'Falsey'
        layout = need_data['layout']  # why are we assigning this? we *just* established that it's Falsey!
```